packages/worker/src/worker-options.ts

```diff
@@ -210,19 +280,37 @@ export interface WorkerOptions {
   /**
    * The number of Workflow isolates to keep in cached in memory
    *
-   * Cached Workflows continue execution from their last stopping point.
-   * If the Worker is asked to run an uncached Workflow, it will need to replay the entire Workflow history.
-   * Use as a dial for trading memory for CPU time.
+   * Cached Workflows continue execution from their last stopping point. If the Worker is asked to run an uncached
+   * Workflow, it will need to fetch and replay the entire Workflow history.
+   *
+   * **When `reuseV8Context` is disabled** — Most users are able to fit at least 250 Workflows per GB of available
+   * memory. The major factors contributing to a Workflow's memory weight are the size of allocations made by the
+   * Workflow itself and by all loaded librairies (including the Node JS builtin context), as well as the size of the
+   * Workflow bundle (code and source map). For the SDK test Workflows, we managed to fit 750 Workflows per GB. Your
+   * millage may vary.
    *
-   * Most users are able to fit at least 250 Workflows per GB of available memory.
-   * The major factors contributing to a Workflow's memory weight are the size of allocations made
-   * by the Workflow itself and the size of the Workflow bundle (code and source map).
-   * For the SDK test Workflows, we managed to fit 750 Workflows per GB.
+   * **When `reuseV8Context` is enabled** — Since most objects are shared/reused across Workflows, the per-Workflow
+   * memory footprint is much smaller. The major factor contributing to a Workflow's memory weight is the size of
+   * allocations made by the Workflow itself. Most users are able to fit at least 600 Workflows per GB of available
+   * memory. In one reference performance test, memory usage grew by approximately 1 MB per cached Workflow (that is
+   * including memory used for activity executions of these Workflows). Your millage may vary.
    *
-   * @default `max(maxHeapMemory / 1GiB - 1, 1) * 250`
+   * @default if `reuseV8Context = true`, then `max(floor(max(maxHeapMemory - 200MB, 0) * (600WF / 1024MB)), 10)`.
+   *          Otherwise `max(floor(max(maxHeapMemory - 400MB, 0) * (250WF / 1024MB)), 10)`
    */
   maxCachedWorkflows?: number;
 
+  /**
+   * Controls the number of Worker threads the Worker should create.
+   *
+   * Threads are used to create {@link https://nodejs.org/api/vm.html | vm }s for the isolated Workflow environment.
+   *
+   * New Workflows are created on the thread having the least active Workflows.
+   *
+   * @default 1 for reuseV8Context, otherwise 8. Ignored if `debugMode` is enabled.
+   */
+  workflowThreadPoolSize?: number;
```

Comment:
Remind me again why we're exposing this?
Have you seen this have any effect on perf?

Reply:
Also, I'm thinking the default of 8 is too high without reuse context. Need to test that.

Reply:
> Remind me again why we're exposing this? Have you seen this have any effect on perf?

Oh... I thought I had documented it, but yes, I've seen positive effect in tests.

With `reuseV8Context`, increasing this to 2 was a must to avoid WF Task Timeouts when setting `maxConcurrentWorkflowTaskExecutions` to higher values, and assuming WF poller was also high enough to keep execution slots filled (ie. we keep the pipeline enough so that WFT handling is CPU bound rather than network bound). At lower value, increasing the number of thread would have no effect (neither negative nor positive), since WFT handling is network bound.

Another reason for exposing this is for cases where people are running several WF Workers from a single node process, with `reuseV8Context` disabled, on rather modest machines. In that case, running 8 threads per worker uselessly creates resource contention. Recently had to recommend one customer to reduce this value on some workers, even though it was technically not exposed.

Reply:
Still 8 is excessive AFAICT, let's lower to 2 by default for non reuse.